brouter-core/src/main/java/btools/router/RoutingEngine.java

```diff
@@ -995,7 +995,7 @@ private OsmTrack tryFindTrack(OsmTrack[] refTracks, OsmTrack[] lastTracks) {
       } else {
         seg = searchTrack(matchedWaypoints.get(i), matchedWaypoints.get(i + 1), i == matchedWaypoints.size() - 2 ? nearbyTrack : null, refTracks[i]);
         wptIndex = i;
-        if (engineMode == BROUTER_ENGINEMODE_ROUNDTRIP) {
+        if (routingContext.continueStraight) {
```

Comment:
@afischerdev Here to solve the problem in https://github.com/abrensch/brouter/pull/774#issuecomment-2780728835, we could use:
```java
if (engineMode == BROUTER_ENGINEMODE_ROUNDTRIP || routingContext.continueStraight) {
```

Reply:
That is not a bug it's a feature. The idea was to make the `continueStraight` switchable and this should also effect the round trips. Otherwise you alway have on round trip this feature on.

I made around 200 random round trips with continueStraight on. 
And again 100 random with off. 
And 100 with undefined value. No problems found.

Reply:
And please notice this:
> ParseException ... profile.brf ...

This is thrown before routing.

Reply:
@afischerdev Have you tested on Android?

The following bundle does not work with `continueStraight=1`.
If we remove the `continueStraight=1` then it works.

```java
Bundle[{exportWaypoints=1, timode=2, allowSamewayback=0, roundTripPoints=5, profile=trekking, alternativeidx=0, roundTripDistance=6333, lonlats=1.384443,43.444626, trackFormat=json, maxRunningTime=600, extraParams=continueStraight=1@correctMisplacedViaPoints=1&correctMisplacedViaPointsDistance=0&allow_steps=1&allow_ferries=1&ignore_cycleroutes=0&stick_to_cycleroutes=0&avoid_unsafe=0&consider_noise=0&consider_river=0&consider_forest=1&consider_town=0&consider_traffic=0&consider_elevation=0, engineMode=4}]
```

Reply:
Please look here:
> ...extraParams=continueStraight=1@correctMisplacedViaPoints=1...

separator should be a `&`

Reply:
@afischerdev What a silly mistake, sorry for the confusion and thanks! :slightly_smiling_face: